### PR TITLE
SW-5681 Added application modules and deliverables endpoint

### PR DESCRIPTION
**Pull request by tommylau523**

### CODE REVIEW

**Comment by tommylau523** [comment]
* **#2198** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2198?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2197** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2197?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2194** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2194?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @tommylau523 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/main/kotlin/com/terraformation/backend/accelerator/api/ApplicationsController.kt [hunk 1]
```diff
@@ -309,6 +408,17 @@ data class GetApplicationResponsePayload(val application: ApplicationPayload) :
 data class ListApplicationsResponsePayload(val applications: List<ApplicationPayload>) :
     SuccessResponsePayload
 
+data class GetApplicationModulesResponsePayload(val modules: List<ApplicationModulePayload>) :
+    SuccessResponsePayload
+
+data class GetApplicationDeliverablesResponsePayload(
```
Comment (nickgraz) [inline src/main/kotlin/com/terraformation/backend/accelerator/api/ApplicationsController.kt:414]: Can these be combined? Lines 414 - 420

File: src/main/kotlin/com/terraformation/backend/accelerator/api/DeliverablesController.kt [hunk 2]
```diff
@@ -66,6 +66,11 @@ class DeliverablesController(
               "accelerator admin privileges, this will be the full list of all deliverables for " +
               "all accelerator projects.")
   fun listDeliverables(
+      @Parameter(
+          description =
+              "Filter deliverables for by modules. Can we used with other request params.")
```
Comment (nickgraz) [inline src/main/kotlin/com/terraformation/backend/accelerator/api/DeliverablesController.kt]:
```suggestion
              "Filter deliverables for by modules. Can be used with other request params.")
```

**Review by nickgraz — APPROVED** [review]
This looks good. If you think both endpoints will always return the same shape of data, I think the `GetApplicationDeliverablesPayload` can be used in both cases (by module or by application). OK to keep them separated, too

**Comment by tommylau523** [comment]
### Merge activity

* **Jul 24, 4:42 PM EDT**: @tommylau523 merged this pull request with [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/2197).

**Pull request merged**